modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataLayoutResourceImpl.java

```diff
@@ -443,6 +444,13 @@ private Page<DataLayout> _getDataLayouts(
 
 		if (Validator.isNull(keywords)) {
 			return Page.of(
+				HashMapBuilder.<String, Map<String, String>>put(
+					"createBatch",
+					addAction(
+						ActionKeys.VIEW, ddmStructure.getStructureId(),
+						"postDataDefinitionDataLayoutBatch",
+						_dataDefinitionModelResourcePermission)
+				).build(),
```

Comment:
@locpham97 @TaiPhamLR shouldn't we add it [here](https://github.com/liferay-forms/liferay-portal/blob/c35f6301888b9b9fbda2b07e50b6a8fc4290ab6a/modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataLayoutResourceImpl.java#L472-L473) too?

Reply:
Thanks for reporting this, @TaiPhamLR.

Could you please try with:

```
DataDefinitionContentType dataDefinitionContentType =
	_dataDefinitionContentTypeRegistry.getDataDefinitionContentType(
		ddmStructure.getClassNameId());

if (Validator.isNull(keywords)) {
	return Page.of(
		HashMapBuilder.<String, Map<String, String>>put(
			"createBatch",
			() -> {
				if (dataDefinitionContentType == null) {
					return null;
				}

				return addAction(
					DDMActionKeys.ADD_STRUCTURE,
					"postDataDefinitionDataLayoutBatch",
					dataDefinitionContentType.getPortletResourceName(),
					contextCompany.getCompanyId());
			}
		).build(),
                 ....
}

return SearchUtil.search(
	HashMapBuilder.<String, Map<String, String>>put(
		"createBatch",
		() -> {
			if (dataDefinitionContentType == null) {
				return null;
			}

			return addAction(
				DDMActionKeys.ADD_STRUCTURE,
				"postDataDefinitionDataLayoutBatch",
				dataDefinitionContentType.getPortletResourceName(),
				contextCompany.getCompanyId());
				}
			).build(),
         .....
```

I believe you should test only with Journal and DocumentLibrary, as Dynamic Data Lists does not use Data Engine